src/world/common/UnkFunc29.inc.c

```diff
@@ -0,0 +1,45 @@
+#include "common.h"
+#include "map.h"
+
+ApiStatus N(UnkFunc29)(ScriptInstance *script, s32 isInitialCall) {
+    s32 i;
+
+    if (isInitialCall) {
+        set_background_color_blend(0, 0, 0, 0xFF);
+        script->functionTemp[0].s = 0xFF;
+        script->functionTemp[1].s = 0;
+    }
+
+    script->functionTemp[0].s -= 8;
+    if (script->functionTemp[0].s < 0) {
+        script->functionTemp[0].s = 0;
+    }
+
+    set_background_color_blend(0, 0, 0, script->functionTemp[0].s);
+    if ((script->functionTemp[0].s == 0) && (script->functionTemp[1].s == 0)) {
+        script->functionTemp[1].s = 1;
+    } else if (script->functionTemp[1].s == 1) {
+        func_8011D82C(0);
+        *D_801512F0 = 0;
+
+        for (i = 0; i < 64; i++) {
+            Npc* npc = get_npc_by_index(i);
+            if (npc != NULL && npc->flags != 0) {
+                if (npc->npcID != -4 && npc->npcID != UNK_ALPHA_FUNC_NPC) {
```

Comment:
The use of hardcoded NPC IDs here is cursed; how often is this function included?

Reply:
17 times. I didn't create that UNK_ALPHA_FUNC_NPC variable though, Ethan already had it in all of these files and its use here matches what he uses it for.

Reply:
Oh, I missed the minus sign, lol. Could you replace the -4 with `NPC_PARTNER` then?

Reply:
Is `UNK_ALPHA_FUNC_NPC` negative?

Reply:
Always positive. Valuse are all in the 5 - 25 range.